### PR TITLE
[Caffe2] Fix missing include dirs of gflags and glog

### PR DESCRIPTION
**Pull request by pjh5**
We didn't see this earlier because these include dirs coincide with those of opencv and other packages, so including those packages covers the include dirs of gflags and glog.

Also changing gflags to gflags::gflags for consistency with the other imported targets. EDIT actually I removed this change.

### CODE REVIEW

File: cmake/Dependencies.cmake [hunk 1]
```diff
@@ -153,6 +154,7 @@ if(USE_GLOG)
   include(${CMAKE_CURRENT_LIST_DIR}/public/glog.cmake)
   if (TARGET glog::glog)
     set(CAFFE2_USE_GOOGLE_GLOG 1)
+    include_directories(SYSTEM ${GLOG_INCLUDE_DIR})
```
[the remark this answers is not in the document]
Comment (pjh5) [reply cmake/Dependencies.cmake]: From what I could tell from the documentation, yes. But it didn't seem to be working. Here's the line that does it https://github.com/pytorch/pytorch/blob/master/caffe2/CMakeLists.txt#L201 . I added an explicit target_link_libraries(caffe2 PUBLIC gflags::gflags) too but that also didn't work either.

[the remark this answers is not in the document]
Comment (ezyang) [reply cmake/Dependencies.cmake]: So... if we are actually going to use new style targets everywhere in the cmake, inability to debug problems like this is a serious problem. (And I agree with you, it is very non-obvious how to debug problems like this; I feel it is one of the reasons why the stone-age style of smashing include paths into global variables has been around so much in cmake). IIRC, targets have "properties" which are queryable. Can we check if the property has the contents we expect?

CC @Yangqing

[the remark this answers is not in the document]
Comment (pjh5) [reply cmake/Dependencies.cmake]: The gflags target has the correct INTERFACE_INCLUDE_DIRECTORIES , checked by calling print_target_properties on it after it was created. According to https://cmake.org/cmake/help/v3.0/prop_tgt/INTERFACE_INCLUDE_DIRECTORIES.html this is the exact property which target_link_libraries should work with. Btw I am on cmake version 3.11.2, so this functionality should definitely exist.

**Comment by pjh5** [comment]
These changes were merged in #8467

**Pull request closed without merging**